lib/maths/CTimeSeriesModel.cc

```diff
@@ -2410,14 +2418,22 @@ void CMultivariateTimeSeriesModel::debugMemoryUsage(core::CMemoryUsage::TMemoryU
     mem->setName("CUnivariateTimeSeriesModel");
     core::CMemoryDebug::dynamicSize("m_Controllers", m_Controllers, mem);
     core::CMemoryDebug::dynamicSize("m_TrendModel", m_TrendModel, mem);
+    // We made various memory improvements in 6.4 partly in preparation for multi-bucket
+    // analysis. This is not going to be available until 6.5; however, we will account for
+    // its memory now. The memory usage is used, for example, to allocate jobs to nodes and
+    // to decide if there is sufficient memory to create jobs. Operationally, we therefore
+    // want to avoid unnecessary changes in model memory between versions. An extra residual
+    // model is a good proxy for the amount of memory multi-bucket will consume.
     core::CMemoryDebug::dynamicSize("m_ResidualModel", m_ResidualModel, mem);
+    core::CMemoryDebug::dynamicSize("m_ResidualModelPad", m_ResidualModel, mem);
     core::CMemoryDebug::dynamicSize("m_AnomalyModel", m_AnomalyModel, mem);
     core::CMemoryDebug::dynamicSize("m_SlidingWindow", m_SlidingWindow, mem);
 }
 
 std::size_t CMultivariateTimeSeriesModel::memoryUsage() const {
     return core::CMemory::dynamicSize(m_Controllers) +
            core::CMemory::dynamicSize(m_TrendModel) +
+           /*see above*/
            core::CMemory::dynamicSize(m_ResidualModel) +
```

Comment:
Should it be `2 *` here as well?

Reply:
Doh! It should indeed.

Reply:
All tests pass. This indicates we should be testing for agreement of `debugMemoryUsage` and `memoryUsage` in some extra cases and ideally also checking partition and by field memory usage in these cases as well. I think I'd prefer to introduce further testing in a follow up PR on master + 6.5.